internal/cmd/base/initial_resources.go

```diff
@@ -584,3 +586,60 @@ func (b *Server) CreateInitialTarget(ctx context.Context) (target.Target, error)
 
 	return tt, nil
 }
+
+func (b *Server) CreateHostPlugin(ctx context.Context, pluginId string, plg plgpb.HostPluginServiceServer) (*hostplugin.Plugin, error) {
```

Comment:
Good point -- I had figured this might be more general purpose at some point, but there's actually no reason to not just do that now. So I've modified it to take in options as you suggested and it now serves as a more general purpose function.